command/healthcheck/pki_crl_validity_period.go

```diff
@@ -0,0 +1,205 @@
+}
+
+func (h *CRLValidityPeriod) IsEnabled() bool {
+	return h.Enabled
+}
+
+func (h *CRLValidityPeriod) DefaultConfig() map[string]interface{} {
+	return map[string]interface{}{
+		"crl_expiry_pct_critical":       "95",
+		"delta_crl_expiry_pct_critical": "95",
+	}
+}
+
+func (h *CRLValidityPeriod) LoadConfig(config map[string]interface{}) error {
+	value, err := parseutil.SafeParseIntRange(config["crl_expiry_pct_critical"], 1, 99)
+	if err != nil {
+		return fmt.Errorf("error parsing %v.crl_expiry_pct_critical=%v: %w", h.Name(), config["crl_expiry_pct_critical"], err)
+	}
+	h.CRLExpiryPercentage = int(value)
+
+	value, err = parseutil.SafeParseIntRange(config["delta_crl_expiry_pct_critical"], 1, 99)
+	if err != nil {
+		return fmt.Errorf("error parsing %v.delta_crl_expiry_pct_critical=%v: %w", h.Name(), config["delta_crl_expiry_pct_critical"], err)
+	}
+	h.DeltaCRLExpiryPercentage = int(value)
+
+	enabled, err := parseutil.ParseBool(config["enabled"])
+	if err != nil {
+		return fmt.Errorf("error parsing %v.enabled: %w", h.Name(), err)
+	}
+	h.Enabled = enabled
+
+	return nil
+}
+
+func (h *CRLValidityPeriod) FetchResources(e *Executor) error {
+	exit, _, issuers, err := pkiFetchIssuers(e, func() {
+		h.UnsupportedVersion = true
+	})
+	if exit {
+		return err
+	}
+
+	for _, issuer := range issuers {
+		exit, _, crl, err := pkiFetchIssuerCRL(e, issuer, false, func() {
+			h.UnsupportedVersion = true
+		})
+		if exit {
+			if err != nil {
+				return err
+			}
+			continue
+		}
+
+		h.CRLs[issuer] = crl
+
+		exit, _, delta, err := pkiFetchIssuerCRL(e, issuer, true, func() {
+			h.NoDeltas = true
+		})
+		if exit {
+			if err != nil {
+				return err
+			}
+			continue
+		}
+
+		h.DeltaCRLs[issuer] = delta
+	}
+
+	// Check if the issuer is fetched yet.
+	configRet, err := e.FetchIfNotFetched(logical.ReadOperation, "/{{mount}}/config/crl")
+	if err != nil {
+		return err
+	}
+
+	h.CRLConfig = configRet
+
+	return nil
+}
+
+func (h *CRLValidityPeriod) Evaluate(e *Executor) (results []*Result, err error) {
+	if h.UnsupportedVersion {
+		ret := Result{
+			Status:   ResultInvalidVersion,
+			Endpoint: "/{{mount}}/issuers",
+			Message:  "This health check requires Vault 1.11+ but an earlier version of Vault Server was contacted, preventing this health check from running.",
+		}
+		return []*Result{&ret}, nil
+	}
+
+	now := time.Now()
+	crlDisabled := false
+	if h.CRLConfig != nil {
+		if h.CRLConfig.IsSecretPermissionsError() {
+			ret := Result{
+				Status:   ResultInsufficientPermissions,
+				Endpoint: "/{{mount}}/config/crl",
+				Message:  "This prevents the health check from seeing if the CRL is disabled and dropping the severity of this check accordingly.",
+			}
+
+			if e.Client.Token() == "" {
+				ret.Message = "No token available so unable read authenticated CRL configuration for this mount. " + ret.Message
+			} else {
+				ret.Message = "This token lacks so permission to read the CRL configuration for this mount. " + ret.Message
```

Comment:
I accidentally deleted this comment from @cipherboy, adding it back.
 
```suggestion
				ret.Message = "This token lacks permission to read the CRL configuration for this mount. " + ret.Message
```